ext/groonga/rb-grn-table.c

```diff
@@ -2678,6 +2678,60 @@ rb_grn_table_rename (VALUE self, VALUE rb_name)
     return self;
 }
 
+/*
+ * Recreates all index columns in the table.
+ *
+ * This method is useful when you have any broken index columns in
+ * the table. You don't need to specify each index column. But this
+ * method spends more time rather than you specify only reindex
+ * needed index columns.
+ *
+ * @example How to recreate all index columns in the table
+ *   Groonga::Schema.define do |schema|
+ *     schema.create_table("Memos") do |table|
+ *       table.short_text("title")
+ *       table.text("content")
+ *     end
+ *
+ *     schema.create_table("Terms",
+ *                         :type => :patricia_trie,
+ *                         :key_type => :short_text,
+ *                         :normalizer => "NormalizerAuto",
+ *                         :default_tokenizer => "TokenBigram") do |table|
+ *       table.index("Memos.title")
+ *       table.index("Memos.content")
+ *     end
+ *   end
+ *
+ *   Groonga["Terms"].reindex
+ *   # =>
+ *   #   Groonga["Terms.Memos_title"].reindex
+ *   #   Groonga["Terms.Memos_content"].reindex
+ *
```

Comment:
The example should show that index columns in the table are only reindex targets. In other words, the example should show that index columns in other table aren't reindex targets explicitly.

```ruby
Groonga::Schema.define do |schema|
  schema.create_table("Memos") do |table|
    table.short_text("title")
    table.text("content")
  end

  schema.create_table("BigramTerms",
                      :type => :patricia_trie,
                      :key_type => :short_text,
                      :normalizer => "NormalizerAuto",
                      :default_tokenizer => "TokenBigram") do |table|
    table.index("Memos.title")
    table.index("Memos.content")
  end

  schema.create_table("MeCabTerms",
                      :type => :patricia_trie,
                      :key_type => :short_text,
                      :normalizer => "NormalizerAuto",
                      :default_tokenizer => "TokenMecab") do |table|
    table.index("Memos.title")
    table.index("Memos.content")
  end
end

Groonga["BigramTerms"].reindex
# =>
#   Groonga["BigramTerms.Memos_title"].reindex
#   Groonga["BigramTerms.Memos_content"].reindex
#
# They aren't called:
#   Groonga["MeCabTerms.Memos_title"].reindex
#   Groonga["MeCabTerms.Memos_content"].reindex
```